data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusSinkBufferEntry.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.prometheus.service;
+
+import org.opensearch.dataprepper.common.sink.SinkBufferEntry;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.metric.JacksonExponentialHistogram;
+import org.opensearch.dataprepper.model.metric.JacksonGauge;
+import org.opensearch.dataprepper.model.metric.JacksonMetric;
+import org.opensearch.dataprepper.model.metric.JacksonHistogram;
+import org.opensearch.dataprepper.model.metric.JacksonSum;
+import org.opensearch.dataprepper.model.metric.JacksonSummary;
+
+public class PrometheusSinkBufferEntry implements SinkBufferEntry {
+
+    final Event event;
+    final PrometheusTimeSeries timeSeries;
+
+    public PrometheusSinkBufferEntry(final Event event, final boolean sanitizeNames) throws Exception {
+        this.event = event;
+        timeSeries = getTimeSeriesForEvent(sanitizeNames);
+    }
+
+    public PrometheusTimeSeries getTimeSeries() {
+        return timeSeries;
+    }
+
+    @Override
+    public long getEstimatedSize() {
+        return timeSeries.getSize();
+    }
+
+    @Override
+    public boolean exceedsMaxEventSizeThreshold() {
+        return false;
+    }
+
+    @Override
+    public Event getEvent() {
+        return event;
+    }
+
+    private PrometheusTimeSeries getTimeSeriesForEvent(final boolean sanitizeNames) throws Exception {
+        if (event.getMetadata().getEventType().equals("METRIC")) {
+            try {
+                PrometheusTimeSeries timeSeries = new PrometheusTimeSeries((JacksonMetric)event, sanitizeNames);
+                if (event instanceof JacksonGauge) {
```

Comment:
Do not rely on the `JacksonX` implementations. Use the interfaces directly. e.g. `JacksonGauge` -> `Gauge`.